aws-gitlab-main.yml

```diff
@@ -0,0 +1,34 @@
+Resources:
+#ROLE W/ IMBEDDED POLICY
+  GitLabS3Access: #Name of the role 
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument: 
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com #?
+            Action:
+              - 'sts:AssumeRole' #??
+      Policies:
+          - PolicyName: gl-s3-policy
+            PolicyDocument:
+              Statement:
+                - Effect: Allow
+                  Action:
+                  - 's3:AbortMultipleUpload'
+                  - 's3:CompletemultipartUpload'
+                  - 's3:ListBucket'
+                  - 's3:PutObject'
+                  - 's3:GetObject'
+                  - 's3:DeleteObject'
+                  - 's3:PutObjectAcl' 
+                  Resource: 'arn:aws:s3:::aws-gitlab-bucket'
```

Comment:
where is this bucket created? can we include it as part of the template and do !Ref , if not parameterized the name

Reply:
I created it in my AWS. Believe I can change it to !Ref. Would I create the bucket with no specified properties? If you wanted to give a BucketName how would you know if the name is taken or not? What do you mean by parameterized the name?

Reply:
If bucket can be created inside the yaml template do that and do not do it in console. Once the final template is pushed the whole thing should go live like magic :) 

Bucket name: 
1. Either make the name dynamic with  unique element  such as !Sub gitlab-bucket-${AWS::AccountId} << unique element means nobody using it. Or do not give bucket a name in the yaml and aws will generate random string for it using LogicalResourceName of the bucket + random string

Bucket will have properties that is encrypting it properly. Potentially versioning and logging enabled depends. you can start with simple bucket w/ default encryption than go more on it